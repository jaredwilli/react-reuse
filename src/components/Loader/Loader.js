import React from 'react';
import PropTypes from 'prop-types';

const Loader = props => {
    return (
        <svg
            width="22px"
            height="22px"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 100 100"
            preserveAspectRatio="xMidYMid">
            <circle
                cx="50"
                cy="50"
                fill="none"
                stroke={props.stroke}
                strokeWidth="10"
                r="45"
                strokeDasharray="212.05750411731105 72.68583470577035"
                transform="rotate(276 50 50)">
                <animateTransform
                    attributeName="transform"
                    type="rotate"
                    calcMode="linear"
                    values="0 50 50;360 50 50"
                    keyTimes="0;1"
                    dur="1s"
                    begin="0s"
                    repeatCount="indefinite">
                </animateTransform>
            </circle>
        </svg>
    );
};

Loader.defaultProps = {
    stroke: '#29292A'
};

Loader.propTypes = {
    stroke: PropTypes.string
};

export default Loader;
