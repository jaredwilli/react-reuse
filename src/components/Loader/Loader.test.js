import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Loader from './Loader';

describe('Loader', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<Loader />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
