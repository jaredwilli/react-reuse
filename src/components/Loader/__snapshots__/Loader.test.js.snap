// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders the component correctly 1`] = `
<svg
  height="22px"
  preserveAspectRatio="xMidYMid"
  viewBox="0 0 100 100"
  width="22px"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="50"
    cy="50"
    fill="none"
    r="45"
    stroke="#29292A"
    strokeDasharray="212.05750411731105 72.68583470577035"
    strokeWidth="10"
    transform="rotate(276 50 50)"
  >
    <animateTransform
      attributeName="transform"
      begin="0s"
      calcMode="linear"
      dur="1s"
      keyTimes="0;1"
      repeatCount="indefinite"
      type="rotate"
      values="0 50 50;360 50 50"
    />
  </circle>
</svg>
`;
